### ISSUE TITLE
BUG 

### ISSUE THREAD

**Issue by ROSHTU** [microsoft/vscode-cpptools#8679]
Existing issue posted by me on github https://github.com/microsoft/vscode/issues/140705#issue-1103610189

We have written the
Issue Type: <b>Bug</b>

int x;
printf("Enter the size : ");
scanf("%d",&x);
// the below error is a bug in vs code extension it does not harm the functioning of the program
bd b[x];
on writing the above given code i found out that a red line appears on x in bd b[x]; which denotes an error but on running the program it does not disrupts the functioning of the program

Extension version: 1.8.0-insiders2
VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:40:02.816Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.83GB (26.21GB free)|
|Process Argv|--crash-reporter-id 53e61994-1cd4-451a-a4ff-08743a6389ee|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
vscop804:30404766
vscop453:30404998
vsrem710:30416614
vsccppwtstudy:30421978

```

</details>

needed data into your clipboard because it was too large to send. Please paste.

**Comment by sean-mcmanus** [comment]
This is a duplicate of https://github.com/microsoft/vscode-cpptools/issues/8581 and should be fixed with our pending 1.8.0-insiders3 (or by downgrading to 1.7.1). Also, you should be able to work around it by adding a compilerArg of -fno-ms-extensions.

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.8.0-insiders3 or 1.8.0 (via using VS Code's "Pre-Release Version" UI or https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools&ssr=false#version-history ).

Unless you're explicitly adding -fms-extensions, in which case that bug fix is still pending (https://github.com/microsoft/vscode-cpptools/issues/8353 ).
